fss_check/check_fss.py

```diff
@@ -0,0 +1,503 @@
+    beta_sun = dat["beta_sun"]
+
+    for i, title, xlabel, ylabel in (
+        (
+            1,
+            "Pitch for alpha error > {} deg".format(angle_err_lim),
+            None,
+            "Pitch (deg)",
+        ),
+        (2, "Pitch when alpha sun presence is False", None, "Pitch (deg)"),
+        (3, "Pitch when beta sun presence is False", None, "Pitch (deg)"),
+    ):
+        plt.figure(i)
+        plt.clf()
+        plt.grid()
+        plt.title(title)
+        plt.ylabel(ylabel)
+        if xlabel:
+            plt.xlabel(xlabel)
+
+    zipvals = zip(
+        (~dat["kalman"], dat["kalman"]),
+        (
+            dict(color="c", mec="c"),  # Not Kalman, No sun presence
+            dict(color="r", mec="r"),
+        ),  # Kalman, No sun presence
+        (
+            dict(color="b", mec="b", fmt="o"),  # Not Kalman, Sun presence
+            dict(color="r", mec="r", fmt="x", mew=2),
+        ),  # Kalman, Sun presence
+        ("Not Kalman (cyan)", "Kalman (red)"),
+    )
+    sun_presence = alpha_sun & beta_sun
+    bad_value = abs(roll_err) > angle_err_lim
+    for filt, opt1, opt2, label in zipvals:
+        plt.figure(1)
+        ok = filt & bad_value
+        if np.any(ok):
+            plot_cxctime(times[ok], pitch[ok], ",", label=label, **opt1)
+
+        ok = filt & sun_presence & bad_value
+        if np.any(ok):
+            plot_cxctime(
+                times[ok], pitch[ok], label=label + " & Sun Presence True", **opt2
+            )
+
+        plt.figure(2)
+        ok = filt & ~alpha_sun
+        if np.any(ok):
+            plot_cxctime(times[ok], pitch[ok], ",", label=label, **opt1)
+
+        plt.figure(3)
+        ok = filt & ~beta_sun
+        if np.any(ok):
+            plot_cxctime(times[ok], pitch[ok], ",", label=label, **opt1)
+
+    suffs = ("bad_alpha_sun", "alpha_no_sun", "beta_no_sun")
+    for i, suff in enumerate(suffs):
+        plt.figure(i + 1)
+
+        set_plot_limits(start, stop)
+
+        plt.legend(loc="best")
+
+        if savefig:
+            ident = savefig if isinstance(savefig, str) else ""
+            plt.savefig("pitch_" + ident + suff + start_suffix + ".png")
+
+
+def plot_delta_vs_pitch_roll(
+    dat: Table,
+    max_pitch: float = 135.0,
+    err_lim: float = 2.0,
+    outfile: Optional[str] = None,
+):
+    """Plot delta pitch and delta roll vs. pitch and roll.
+
+    This results in a 2x2 plot with the following subplots:
+    - delta pitch vs. pitch
+    - delta roll vs. pitch
+    - delta pitch vs. roll
+    - delta roll vs. roll
+
+    Parameters
+    ----------
+    dat : astropy.table.Table
+        Table of FSS data
+    outfile : str, optional
+        Filename to write plot to.
+    """
+    logger.info(f"{max_pitch=} {err_lim=} {outfile=}")
+    marker = "."
+    marker_size = 1
+
+    ok_pitch = dat["pitch"] < max_pitch
+    _, ((ax0, ax1), (ax2, ax3)) = plt.subplots(2, 2, figsize=(8, 6))
+
+    def plot_one(ax, xaxis, yaxis):
```

Comment:
I think this is fine, but this plot method doesn't have a really helpful name.

Reply:
Added a docstring. The function name matters less since this is an encapsulated function.